File: apps/manpou-allinone/src/main/java/com/manpou/allinone/offlineorder/domain/model/OfflineOrder.java
package com.manpou.allinone.offlineorder.domain.model;

import jakarta.persistence.Access;
import jakarta.persistence.AccessType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 线下订单实体，映射遗留表 list6。
 */
@Entity
@Table(name = "list6")
@Access(AccessType.FIELD)
@Getter
@Setter
public class OfflineOrder {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Long id;

    @Column(name = "showFlag")
    private Integer showFlag;

    @Column(name = "code", length = 20)
    private String code;

    @Column(name = "sub_code", length = 20)
    private String subCode;

    @Column(name = "houkoku", length = 50)
    private String houkoku;

    @Column(name = "info_file", length = 10)
    private String infoFile;

    @Column(name = "item_name", length = 50)
    private String itemName;

    @Column(name = "volume_count")
    private Integer volumeCount;

    @Column(name = "order_count")
    private Integer orderCount;

    @Column(name = "expected_date")
    private LocalDate expectedDate;

    @Column(name = "order_date")
    private LocalDate orderDate;

    @Column(name = "arrival", length = 10)
    private String arrival;

    @Column(name = "unit_ch")
    private Double unitCh;

    @Column(name = "rate")
    private Double rate;

    @Column(name = "souko", length = 10)
    private String souko;

    @Column(name = "factory", length = 50)
    private String factory;

    @Column(name = "contactor", length = 20)
    private String contactor;

    @Column(name = "contactor_tel", length = 30)
    private String contactorTel;

    @Column(name = "principal", length = 30)
    private String principal;

    @Column(name = "memo", columnDefinition = "text")
    private String memo;

    @Column(name = "link", columnDefinition = "text")
    private String link;

    @Column(name = "updater", length = 20)
    private String updater;

    @Column(name = "updatetime")
    private LocalDateTime updatetime;

    @Column(name = "inventory_note", precision = 10, scale = 0)
    private BigDecimal inventoryNote;

    @Column(name = "rireki", columnDefinition = "longtext")
    private String rireki;
}

File: apps/web/src/api/offline-order.ts
import client from './client'

export interface OfflineOrderPageVO {
  id: number
  showFlag: number
  code: string
  subCode: string
  houkoku: string
  infoFile: string
  itemName: string
  volumeCount: number
  orderCount: number
  expectedDate: string
  orderDate: string
  arrival: string
  unitCh: number
  rate: number
  souko: string
  factory: string
  contactor: string
  contactorTel: string
  principal: string
  memo: string
  link: string
  updater: string
  updatetime: string
  inventoryNote: number
  rireki: string
}

export interface OfflineOrderPageResponse {
  content: OfflineOrderPageVO[]
  totalElements: number
  totalPages: number
  pageNumber: number
}

export interface CreateOfflineOrderRequest {
  showFlag?: number
  code?: string
  subCode?: string
  houkoku?: string
  infoFile?: string
  itemName?: string
  volumeCount?: number
  orderCount?: number
  expectedDate?: string
  orderDate?: string
  arrival?: string
  unitCh?: number
  rate?: number
  souko?: string
  factory?: string
  contactor?: string
  contactorTel?: string
  principal?: string
  memo?: string
  link?: string
  inventoryNote?: number
  rireki?: string
}

export interface UpdateOfflineOrderRequest {
  showFlag?: number
  code?: string
  subCode?: string
  houkoku?: string
  infoFile?: string
  itemName?: string
  volumeCount?: number
  orderCount?: number
  expectedDate?: string
  orderDate?: string
  arrival?: string
  unitCh?: number
  rate?: number
  souko?: string
  factory?: string
  contactor?: string
  contactorTel?: string
  principal?: string
  memo?: string
  link?: string
  inventoryNote?: number
  rireki?: string
}

export const offlineOrderApi = {
  list(params: { page: number; pageSize: number; code?: string; itemName?: string; factory?: string; arrival?: string }) {
    return client.get<OfflineOrderPageResponse>('/offline-orders', { params })
  },
  get(id: number) {
    return client.get<OfflineOrderPageVO>(`/offline-orders/${id}`)
  },
  create(data: CreateOfflineOrderRequest) {
    return client.post<number>('/offline-orders', data)
  },
  update(id: number, data: UpdateOfflineOrderRequest) {
    return client.patch<{ code: string }>(`/offline-orders/${id}`, data)
  },
  delete(id: number) {
    return client.delete<{ code: string }>(`/offline-orders/${id}`)
  },
}